### PR TITLE
fix(Autocomplete): enhance input handling for controlled scenarios

### PR DESCRIPTION
**Pull request by snitin315**
Fix https://github.com/razorpay/blade/issues/1676

### CODE REVIEW

**Review by anuraghazra — APPROVED** [review]
Looks good, possible to add a test?

**Comment by snitin315** [comment]
I believe testing is not required for this.

**Comment by kamleshchandnani** [comment]
> I believe testing is not required for this.

@snitin315 why do you say that? I feel we shall add a test for this to make sure this doesn't occur in future?

**Comment by snitin315** [comment]
I will add one, in this case adding a test case takes more time than writing the fix. Also, it's super rare not to call `setValue` in `onChange`, in controlled scenarios.

File: packages/blade/src/components/Input/DropdownInputTriggers/AutoComplete.tsx [hunk 1]
```diff
@@ -127,7 +127,8 @@ const useAutoComplete = ({
         name: props.name,
         value: displayText,
       });
-      setInputValue(displayText ?? '');
+      // Use displayText as inputValue only if its not controlled by user
+      setInputValue((props.value as string) ?? displayText ?? '');
```
Comment (saurabhdaware) [inline packages/blade/src/components/Input/DropdownInputTriggers/AutoComplete.tsx:131]: displayText is supposed to be different from value.

E.g. if `value="mumbai"`, and we have that value mapped to `"Mumbai"` in ActionListItem, then the value displayed in input is "Mumbai" and not "mumbai".


I think what we need to do here might be to not call `setInputValue` when value is being controlled 🤔

Comment (snitin315) [reply packages/blade/src/components/Input/DropdownInputTriggers/AutoComplete.tsx:131]: Let me check if that fixes the issue.

**Review by saurabhdaware — DISMISSED** [review]
Looks good to me

**Review by saurabhdaware — APPROVED** [review]

**Pull request merged**